### ISSUE TITLE
[CT-1080] BigQuery dataset expiration time

### ISSUE THREAD

**Issue by akashgangulyhf** [dbt-labs/dbt-bigquery#289]
### Describe the feature
BigQuery supports datasets to expire after a given time automatically.
Currently DBT doesn't support configuring the BQ dataset(schema).

### Describe alternatives you've considered
We have to configure and create the dataset first in BQ and then use it form DBT.

### Additional context
Creating the dataset first in BQ doesn't help fully for cases where we would want to dynamically create datasets and expire them automatically on CI pipeline.

### Who will this benefit?
On GitHub PR, we can use this with DBT Core, where we create BQ datasets with a specific expiration time. That way we don't have to give DELETE access to GitHub explicitly.

### Are you interested in contributing this feature?
NA

**Comment by jtcohen6** [comment]
@akashgangulyhf Thanks for opening!

Since https://github.com/dbt-labs/dbt-bigquery/pull/183 was merged, and will be included in v1.3, it will be newly possible for you to define your own `create_schema` macro, and have dbt use that one [instead of the default](https://github.com/dbt-labs/dbt-core/blob/8c8be687019014ced9be37c084f944205fc916ab/core/dbt/include/global_project/macros/adapters/schema.sql#L1-L9):
```sql
{% macro create_schema(relation) %}
  {%- call statement('create_schema') -%}
    create schema if not exists {{ relation.without_identifier() }}
    options (
        default_table_expiration_days = 3600 -- in seconds (1 hour)
    )
  {% endcall %}
{% endmacro %}
```

---

Still, we're missing a mechanism for users to _truly_ configure schemas/datasets in dbt today. We can say that, without necessarily implying that dbt should first / eventually become a totally generic and extensible framework for managing all types of database objects. It's already in the business of creating schemas/datasets, as soon as you define and run a model in a schema/dataset that does not yet exist. This has come up in the context of labels (https://github.com/dbt-labs/dbt-bigquery/issues/22), persisting descriptions/comments (https://github.com/dbt-labs/dbt-core/issues/1714), grants ([dev blog](https://docs.getdbt.com/blog/configuring-grants#granting-permissions-on-other-object-types)), and managing "orphaned" objects (https://github.com/dbt-labs/dbt-core/issues/4957).

@Fleid something I'd be interested in talking about with you more!

**Comment by jtcohen6** [comment]
I'm going to close this issue in favor of a discussion, about how dbt might enable users to manage schemas more naturally in the future: https://github.com/dbt-labs/dbt-core/discussions/5781

I think the manual override of the `create_schema` macro offers a halfway decent workaround in the meantime.
